Find a functionally equivalent code:android.media.MediaFocusControl.removeFocusStackEntry:COMMENT
Method Modifier: private     
Comment:/**
 * Helper function:
 * Called synchronized on mAudioFocusLock
 * Remove a focus listener from the focus stack.
 * @param clientToRemove the focus listener
 * @param signal if true and the listener was at the top of the focus stack, i.e. it was holding
 * focus, notify the next item in the stack it gained focus.
 */

@@ -7,16 +7,12 @@
         if (signal) {
             // notify the new top of the stack it gained focus
             notifyTopOfAudioFocusStack();
-            // there's a new top of the stack, let the remote control know
-            synchronized (mRCStack) {
-                checkUpdateRemoteControlDisplay_syncAfRcs(RC_INFO_ALL);
-            }
         }
     } else {
         // focus is abandoned by a client that's not at the top of the stack,
@@ -19,25 +15,21 @@
         // evaluated it, traversal order doesn't matter here)
         Iterator<FocusRequester> stackIterator = mFocusStack.iterator();
         while (stackIterator.hasNext()) {
-            FocusRequester fr = (FocusRequester) stackIterator.next();
+            FocusRequester fr = stackIterator.next();
             if (fr.hasSameClient(clientToRemove)) {
                 Log.i(TAG, "AudioFocus  removeFocusStackEntry(): removing entry for " + clientToRemove);
                 stackIterator.remove();

